src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -55,7 +57,25 @@
 */
 static char* jsonPayloadClean(const char* payload)
 {
-  return (char*) strstr(payload, "{");
```

Comment:
I understand this has changed due to the old way doesn't work. Why?=

Reply:
The query response of GET /v2/entities is a JSON Vector (the initial '[' was eaten).
Before, in v1 the response was always an Object and the function worked just fine

Reply:
NTC

Reply:
To keep the original implementation (which is simpler) we could use the following:

```
static char* jsonPayloadClean(const char* payload, const &std::string init)
{
    return (char*) strstr(payload, init);
}
```

So you call `jsonPayloadClean(out.c_str(), "[");` or `jsonPayloadClean(out.c_str(), "{");` depending the case.

In fact, I guess it should even better to use as signature

```
static char* jsonPayloadClean(const char* payload, char init)
```

using the strstr() equivalent for char internally.

Reply:
This idea works if we are 100% sure that there is no '[' in the HTTP header information that comes before the payload.
The old implementation was unsafe. The new implementation is not.
Not a good idea, so:

NTC

Reply:
I'm still trying to understand the logic... I guess that that magic number of 13 means `\r`.

I understand you are looking for either a `\n\n` or `\n\r\n` pattern. In that case, why don't just do somethink like this?

```
char* p;
if ( (p = (char*) strstr(payload, "\n\n")) != NULL)
{
   return p;
}
else
{
   return (char*) strstr(payload, "\n\r\n");
}
```

Clearer from my point of view and also 100% safe.

Reply:
Sure, there are a hundred ways to solve a thing like this.
However ... Your solution is no good.
If `strstr(payload, "\n\r\n")` returns NULL, then 3 will be returned ...
Might be that my solution is bad in a similar way.
I'll take a look at it

Reply:
Nope, no problem there. NULL returned if no payload is found.
Added a few comments to make the function even easier to understand, in bf50da234

Reply:
I'm stil thinking that the version above based in two strstr() statements in simpler and shorter (specially if you add +4 lines to the PR implementation, due to style). However, not importar enough to keep this thread opened. I'll have a look and eventally modify it when fowarding works ends.

NTC

Reply:
If you do two calls to `strstr`, you start looking from the beginning again the second time. Bad idea.

NTC